moped-editor/src/views/projects/projectView/ProjectActivityEntry.js

```diff
@@ -10,22 +10,27 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const ProjectActivityEntry = ({
-  changeIcon,
-  changeDescription,
-  changeValue,
-}) => {
+/**
+ * One row/entry in the project activity log
+ * @param {changeIcon} Material UI icon component
+ * @param {changeText} - Array of objects of the shape:
+ *  {text: String to display, style: String name of style or null}
+ */
+const ProjectActivityEntry = ({ changeIcon, changeText }) => {
   const classes = useStyles();
-
   return (
     <Box display="flex" p={0}>
       <Box p={0}>{changeIcon}</Box>
       <Box p={0} flexGrow={1}>
         <Typography variant="body2" className={classes.entryText}>
-          {changeDescription}
-          {!!changeValue && (
-            <span className={classes.boldText}>{changeValue}</span>
-          )}
+          {
+            // maps through the array of objects and applies specified style to the text
+            changeText.map((changeObject) => (
+              <span className={classes[changeObject.style]}>
```

Comment:
i'm getting a warning in the console saying each of these spans should have a unique key

Reply:
I did some digging in the React docs, and they say that using the index as keys is okay as a last resort. The bugs that stem from index as keys only apply to items that change order in the UI. https://reactjs.org/docs/lists-and-keys.html#keys